src/brogue/Items.c

```diff
@@ -3625,7 +3625,6 @@ boolean polymorph(creature *monst) {
 
     // After polymorphing, don't "drop" any creature on death (e.g. phylactery, phoenix egg)
     if (monst->carriedMonster) {
-        killCreature(monst->carriedMonster, true);
         freeCreature(monst->carriedMonster);
```

Comment:
Might it make sense to delete the call to `freeCreature` instead? An "administrative death" seems the right action here from a meaning POV. I just didn't realise the free would happen elsewhere (in the graveyard?) when I implemented it

Reply:
The way I understand it, polymorph isn't a kill so I'm not sure why a call is necessary, especially on the carried monster, which shouldn't get added to the graveyard. It seems the steps for polymorph are basically clearing out the sort of unique properties for monster (the ones not stored in monster.info) before randomly picking a new type and stomping on monster.info. This includes clearing the carried monster, if there is one. Let me know if I'm missing something.

Reply:
Polymorph isn't a kill but if it's carrying an alive monster, then I thought it should be killed, with all the appropriate other checks being done. Why shouldn't it be added to the graveyard? There is a similar kill in the code for multiplicity (search killCreature in Combat.c applyArmorRunicEffect), where the carried monster is killed after cloning for the spectral copy.

However, it does look like killCreature assumes the creature is placed on the map, so I'm not sure if it's really valid to call it on an unplaced monster... This is probably the most correct then